src/open_inwoner/js/components/form/FileInput.js

```diff
@@ -261,53 +275,57 @@ export class FileInput extends Component {
-    // renderFileHTML is a separate function, where the context of 'this' changes when it is called
     const { name, size, type } = file
     const ext = name.split('.').pop().toUpperCase()
     const sizeMB = (size / (1024 * 1024)).toFixed(2)
     const labelDelete = this.getFilesList().dataset.labelDelete || 'Delete'
     const getFormNonFieldError = this.getFormNonFieldError()
     const formSubmitButton = this.getFormSubmitButton()
 
-    // Only show errors notification if data-max-file-size is exceeded + add error class to file-list
     const maxMegabytes = this.getLimit()
+    const uploadFileTypes = this.getUploadTypes().toUpperCase()
+
+    const sizeError = sizeMB > maxMegabytes
+    const typeError = !uploadFileTypes.includes(ext)
 
     const htmlStart = `
-      <li class="file-list__list-item">
-        <aside class="file">
-          <div class="file__container">
-            ${
-              sizeMB > maxMegabytes
-                ? '<div class="file__file error">'
-                : '<div class="file__file">'
-            }
-              <p class="file__symbol">
-                <span aria-hidden="true" class="material-icons-outlined">${
-                  type.match('image') ? 'image' : 'description'
-                }</span>
-              </p>
-              <p class="p file__data">
-                <span class="file__name">${name} (${ext}, ${sizeMB}MB)</span>
-              </p>
-              <a class="link link--primary" href="#" role="button" aria-label="${labelDelete}">
-                <span aria-hidden="true" class="material-icons-outlined">delete</span>
-              </a>
-            </div>
+    <li class="file-list__list-item">
+      <aside class="file">
+        <div class="file__container">
+          <div class="file__file ${typeError || sizeError ? 'error' : ''}">
+            <p class="file__symbol">
+              <span aria-hidden="true" class="material-icons-outlined">${
+                type.match('image') ? 'image' : 'description'
+              }</span>
+            </p>
+            <p class="p file__data">
+              <span class="file__name">${name} (${ext}, ${sizeMB}MB)</span>
+            </p>
+            <a class="link link--primary" href="#document-upload" role="button" aria-label="${labelDelete}">
+              <span aria-hidden="true" class="material-icons-outlined">delete</span>
+            </a>
           </div>
-        </aside>
-      </li>`
-
-    if (sizeMB > maxMegabytes) {
+          ${
+            typeError || sizeError
+              ? `
+                <p class="p p--small error">
+                  <span aria-hidden="true" class="material-icons-outlined">warning_amber</span>
+                  ${
+                    typeError
+                      ? `Dit type bestand ("${ext}") is ongeldig. Geldige bestandstypen zijn: ${uploadFileTypes}.`
+                      : ''
+                  }
+                  ${sizeError ? 'Dit bestand is te groot' : ''}
```

Comment:
Not sure if we consider this an issue, but if a file has both a `sizeError` and a `typeError` the `sizeError` does not appear as a separate error, but is added to the end instead:

![image](https://github.com/maykinmedia/open-inwoner/assets/29249171/39b6cd52-4af3-4a4d-95c2-c84d2cfff342)

Reply:
If I would put them on a separate line, then I'd guess the icon should be added to the second error as well.
Or: It could be more important if the file is uploaded is of the wrong type, and then fileSize errors are not important anymore?
Let me ask the designer.

Reply:
I'd prefer the first option: show both errors at the same time, but on separate lines and each with its own error icon. It's annoying to resolve one error (e.g. convert to different file type), only to then find out that there's another problem.

Reply:
@pi-sigma I've already discussed this with the designer and the municipality; in case of a combined error we will get a (new) combined message with 1 icon only, so I will add that.